File: haskell/p-02.hs
myButLast :: [t] -> t
myButLast [] = error "Empty list"
myButLast [x] = error "List with ony one element"
myButLast (x:y:[]) = x
myButLast (x:y:zs) = myButLast (y:zs)

-- Simple function composition solution
myButLast' :: [c] -> c
myButLast' = head . tail . reverse

File: erlang/p_02.erl
-module(p_02).
-export([my_but_last/1]).

my_but_last([]) ->
    erlang:error(empty);
my_but_last([_]) ->
    erlang:error(only_one_element);
my_but_last([X,_]) ->
    X;
my_but_last([_|T]) ->
    my_but_last(T).

